### ISSUE TITLE
inclusion of NLU Evaluation Corpora

### ISSUE THREAD

**Issue by jdavidthomson** [explosion/spaCy#3033]
Hey,

Just found this repo and noticed that on my version of spacy there are entities that they talk about in the link below that weren't recognized by the version of spacy i tried.

From this file:
https://github.com/sebischair/NLU-Evaluation-Corpora/blob/master/AskUbuntuCorpus.json

Ubuntu wasn't recognized as an entity and 13.10 was only recognized as a CARDINAL.

"text": "upgrading to Ubuntu 13.10 from Ubuntu 13.04",
 "entities": [
                {
                    "text": "13.10",
                    "entity": "UbuntuVersion",
                    "stop": 5,
                    "start": 3
                },
                {
                    "text": "13.04",
                    "entity": "UbuntuVersion",
                    "stop": 10,
                    "start": 8
                }
            ],

https://github.com/sebischair/NLU-Evaluation-Corpora

Can we integrate the sort of data in the NLU-Evaluation-Corpora, into the models?

Thanks!

Best,
Dave

**Comment by ines** [comment]
Yeah, there are always trade-offs and the model's predictions always depend on the data it was trained on and how that data was labelled. For example, the label scheme used in the English NER corpus spaCy's models were trained on (Onto Notes 5) labelled all numbers as `CARDINAL` – which is a pretty good general-purpose solution.

While you can always [update the entity recognizer](https://spacy.io/usage/training#section-ner) with more examples to fine-tune its redictions on a specific text, I'm not sure this would work very well using this NLU evaluation data.

Even if you application's task is to extract versions of Ubunutu, trying to predict a label `UbuntuVersion` as a named entity recognition task is problematic.  One thing that's important to consider is that NLP models can generally only look at a very narrow context window. This is especially true for named entity recognition and short text classification. "Ubunutu version" isn't really a defined "category of things" like person, organization, date or number. It's pretty much impossible to determine based on the local context of a sentence like this: "Upgrading from 11.10 to 12.04" (unless you let you model assume that all numbers that look like that are Ubuntu versions).

If you wanted to actually solve this task (extracting software versions from AskUbuntu posts), a better approach would probably be to fine-tune the already pre-trained `CARDINAL` label on the data and then use a rule-based approach to determine whether it's an Ubuntu version number (normalize the number, check it against a dictionary of existing versions, maybe check the dependency parse to find how it's used in the sentence). For more background on the reasoning behind this, see @honnibal's [talk on solving different NLP problems and designing label schemes](https://www.youtube.com/watch?v=jpWqz85F_4Y&list=PLBmcuObd5An7oE73XacDNJ3ZxO82ogc6j).

That said, if the type of texts in that dataset are similar to what your application has to deal with, you can definitely use it to bootstrap training data. For example, `UbuntuVersion` are all good examples of number entities in very specific contexts. Or the spans annotated as `WebService` [here](https://github.com/sebischair/NLU-Evaluation-Corpora/blob/master/WebApplicationsCorpus.json) are nice examples of products or even a new category like `WEBSITE`. The intents like `"Filter Spam"` could also be useful to train a text classifier.
